src/api/libcellml/analyserequationast.h

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "libcellml/analyser.h"
+#include "libcellml/generator.h"
+
+#ifdef NAN
+#    undef NAN
+#endif
+
+namespace libcellml {
+
+/**
+ * @brief The AnalyserEquationAst class.
+ *
+ * The AnalyserEquationAst class is for representing an equation abstract syntax
+ * tree (AST) in the context of a CellML Analyser.
+ */
+class LIBCELLML_EXPORT AnalyserEquationAst
+{
+    friend class Analyser;
+
+public:
+    enum class Type
```

Comment:
This enumeration needs some Doxygen documentation.

Reply:
Done in commit 40a3682e.